tests/behat/behat_repository_ocis.php

```diff
@@ -94,39 +124,123 @@ private function get_password_for_user(string $username): string {
+        $spaceid = $this->get_all_created_project_space()[0]["spaceId"];
+        $header = ["Purge" => "T"];
+        $disabledrive = $client->request('DELETE', "/graph/v1.0/drives/$spaceid");
+        $deletedrive = $client->request('DELETE', "/graph/v1.0/drives/$spaceid", null, $header);
+        if ($disabledrive['statusCode'] && $deletedrive['statusCode'] !== 204) {
+            throw new Exception('Error deleting drive');
         }
     }
 
     private function delete_file_in_personal_space($user, $file) {
-        $client = $this->get_client($this->get_actual_username($user), $this->get_password_for_user($user));
+        $client = $this->get_client($user);
         $response = $client->request('DELETE', "/dav/files/$user/$file");
         if ($response['statusCode'] !== 204) {
             throw new Exception("Error deleting resource in ocis " . var_dump($response['statusCode']));
         }
     }
+
+    public function create_resource_in_project_space($space, $resource, $body): array {
+        $client = $this->get_admin_client();
+        $spaceid = $this->get_space_id_by_space($space);
+        $response = $client->request('PUT', "dav/spaces/$spaceid/$resource", $body);
+        return $response;
+    }
+
+    /**
+     * @Given I log in to ocis as :user
+     */
+    public function i_log_in_to_ocis($user) {
+        $this->execute('behat_forms::set_field_node_value', [
+            $this->find('xpath', "//*[@id='oc-login-username']"), $this->get_actual_username($user),
+        ]);
+
+        $this->execute('behat_forms::set_field_node_value', [
+            $this->find('xpath', "//*[@id='oc-login-password']"), $this->get_password_for_user($user),
+        ]);
+
+        $loginbutton = $this->get_selected_node("button", "Log in");
+        $this->ensure_node_is_visible($loginbutton);
+        $loginbutton->click();
+
+        $allowbutton = $this->get_selected_node('xpath_element', "//span[text()='Allow']");
+        $this->ensure_node_is_visible($allowbutton);
+        $allowbutton->click();
+        $this->getSession()->switchToWindow(null);
+    }
+
+    /**
+     * @Given :user uploads file with content :content to :file in :space space
+     *
+     * @param string $user
+     * @param string $content
+     * @param string $file
+     * @param string $space
+     */
+    public function user_creates_a_file($user, $content, $file, $space) {
+        if (strtolower($space) === "personal") {
+            $this->create_file_in_personal_space($user, $file);
+        } else {
+            $this->create_resource_in_project_space($space, $file, $content);
+        }
+    }
+
+    /**
+     * @Given :user creates a project space :space
+     *
+     * @param string $user
+     * @param string $space
+     *
+     * @return void
+     */
+    public function user_creates_a_project_space(string $user, string $space) {
+        $this->create_project_space($user, $space);
+    }
+
+    /**
+     * @Given I click on Refresh button
```

Comment:
I think its better to describe the functionality

```suggestion
     * @Given I refresh the file-picker
```